### PR TITLE
Rework --help for `clouds` command

### PR DESCRIPTION
**Pull request by pmatulis**
## Description of change

This improves the help output for the `clouds` command.

I hardcoded the supported private clouds. I did not include 'oci' nor 'oracle' in that list since the latter should soon be gone and the former will soon be baked in.

See #9280.

## QA steps

View the help (`juju help clouds`) and see the revised output:

```
Details:
Output includes fundamental properties for each cloud known to Juju.
These are: name, number of regions, default region, type, and description.

The cloud name is what's used to refer to a cloud in any Juju command.

The `regions` command lists a cloud's regions. Both regions and endpoints
are exposed with `show-cloud`.

The default output shows those clouds known to Juju out of the box (the
"baked in" clouds). With the exception of the 'localhost' cloud, these
are all the supported public clouds. Other (private) clouds need to be
added via the `add-cloud` command. These are 'lxd' (remote), 'maas',
'manual', 'openstack', and 'vsphere'.

Cloud metadata sometimes changes (e.g. AWS adds a new region). To
synchronise Juju with the "baked in" clouds the `update-clouds` command
is used.

This command's default output format is 'tabular'.

Further reading: https://docs.jujucharms.com/stable/clouds
```

Instead of:

```
Details:
Provided information includes 'cloud' (as understood by Juju), cloud
'type', and cloud 'regions'.
The listing will consist of public clouds and any custom clouds made
available through the `juju add-cloud` command. The former can be updated
via the `juju update-clouds` command.
By default, the tabular format is used.
```

## Documentation changes

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1796149

### CODE REVIEW

File: cmd/juju/cloud/list.go [hunk 1]
```diff
@@ -28,17 +28,27 @@ type listCloudsCommand struct {
...
-	"Provided information includes 'cloud' (as understood by Juju), cloud\n" +
-	"'type', and cloud 'regions'.\n" +
-	"The listing will consist of public clouds and any custom clouds made\n" +
-	"available through the `juju add-cloud` command. The former can be updated\n" +
-	"via the `juju update-clouds` command.\n" +
-	"By default, the tabular format is used.\n" + listCloudsDocExamples
+        "Output includes fundamental properties for each cloud known to Juju.\n" +
+        "These are: name, number of regions, default region, type, and description.\n" +
+        "\nThe cloud name is what's used to refer to a cloud in any Juju command.\n" +
+        "\nThe `regions` command lists a cloud's regions. Both regions and endpoints\n" +
```
[the remark this answers is not in the document]
Comment (pmatulis) [reply cmd/juju/cloud/list.go]: Alright. I wanted to point to more information about one of the mentioned properties (regions) but I believe your argument is stronger.

File: cmd/juju/cloud/list.go [hunk 2]
```diff
@@ -28,17 +28,27 @@ type listCloudsCommand struct {
...
+	"are exposed with `show-cloud`.\n" +
+        "\nThe default output shows those clouds known to Juju out of the box (the\n" +
+	"\"baked in\" clouds). With the exception of the 'localhost' cloud, these\n" +
```
[the remark this answers is not in the document]
Comment (pmatulis) [reply cmd/juju/cloud/list.go]: I see that you removed the list of private clouds that can be added. That means that the user has no idea what specific clouds can be added.

[the remark this answers is not in the document]
Comment (anastasiamac) [reply cmd/juju/cloud/list.go]: Yes, i removed the list as I believe this info is more appropriate on add-cloud command. The type list can change and we do not want to sprinkle it too much... There has to be one point of truth.

File: cmd/juju/cloud/list.go [hunk 3]
```diff
@@ -28,17 +28,27 @@ type listCloudsCommand struct {
...
 // commands for ease in markdown.
 var listCloudsDoc = "" +
-	"Provided information includes 'cloud' (as understood by Juju), cloud\n" +
-	"'type', and cloud 'regions'.\n" +
-	"The listing will consist of public clouds and any custom clouds made\n" +
-	"available through the `juju add-cloud` command. The former can be updated\n" +
-	"via the `juju update-clouds` command.\n" +
-	"By default, the tabular format is used.\n" + listCloudsDocExamples
+        "Output includes fundamental properties for each cloud known to the\n" +
+        "current Juju client. These are: name, number of regions, default region,\n" +
```
Comment (anastasiamac) [inline cmd/juju/cloud/list.go]:
```
... Juju client: name, number of regions,...
```
No point in having 2 separate verbose sentences.

File: cmd/juju/cloud/list.go [hunk 4]
```diff
@@ -28,17 +28,27 @@ type listCloudsCommand struct {
...
+	"type, and description.\n" +
+        "\nThe cloud name is what's used to refer to a cloud in any Juju command.\n" +
```
Comment (anastasiamac) [inline cmd/juju/cloud/list.go]: Pretty sure that this sentence is not needed. It's obvious that cloud name is a cloud name.

File: cmd/juju/cloud/list.go [hunk 5]
```diff
@@ -28,17 +28,27 @@ type listCloudsCommand struct {
...
+        "\nThe default output shows public clouds known to Juju out of the box\n" +
+	"(these may change between Juju versions). In addition to these public\n" +
```
Comment (anastasiamac) [inline cmd/juju/cloud/list.go]: Please do not use parenthesis here. It's a first class citizen and an important piece of information.

File: cmd/juju/cloud/list.go [hunk 6]
```diff
@@ -28,17 +28,27 @@ type listCloudsCommand struct {
...
+	"clouds, the 'localhost' cloud (local LXD) is also listed.\n" +
+        "\nThis command's default output format is 'tabular'.\n" +
+        "\nCloud metadata sometimes changes, e.g. AWS adds a new region. Use the\n" +
+	"`update-clouds` command to update the current Juju client accordingly.\n" +
+	"\nUse the `add-cloud` command to add a private cloud to the list of clouds\n" +
+	"known to the current Juju client.\n" +
+        "\nUse the `regions` command to list a cloud's regions and the `show-cloud`\n" +
```
Comment (anastasiamac) [inline cmd/juju/cloud/list.go]: Let's not compound these 2 sentences. I prefer a separate sentence for each command.

**Comment by pmatulis** [comment]
@anastasiamac Here is the updated text: http://paste.ubuntu.com/p/STrWRgyv5P/

I do think that calling out the cloud name as being the main handle ('aws-china' and not 'aws'; 'google' and not 'gce'; 'localhost' and not 'lxd') to use in commands is important but that's just me.

**Comment by anastasiamac** [comment]
> @anastasiamac Here is the updated text: http://paste.ubuntu.com/p/STrWRgyv5P/
> 
> I do think that calling out the cloud name as being the main handle ('aws-china' and not 'aws'; 'google' and not 'gce'; 'localhost' and not 'lxd') to use in commands is important but that's just me.

@pmatulis,
I would have agreed but 'aws' and 'gce' are cloud types and not cloud names, and we are explicit everywhere where the cloud is needed that a cloud name is expected. 
Also, to say "all commands use cloud name" here could be misleading for commands that may not be using cloud name. I'd rather we avoided prescriptive generalizations like this.

**Comment by pmatulis** [comment]
!!build!! Some kind of oddity occured.

**Comment by pmatulis** [comment]
'aws' is actually a cloud name. :wink:

**Comment by pmatulis** [comment]
Thanks for the build help.

I thought 'ec2' is the cloud type for the 'aws' cloud?

**Comment by anastasiamac** [comment]
@pmatulis,
I was just following your example! But yes, in real world, Juju recognizes Amazon clouds as of 'ec2' type and the public cloud name for the general use cloud is 'aws'. This discussion and that example is irrelevant to this PR however.  As mentioned, the enumeration of the supported cloud types does not belong here and re-iteration of 'cloud name is cloud name used in other commands' is not helpful.

**Comment by pmatulis** [comment]
[Rewrite --help for `list-clouds`](https://trello.com/c/sqUEt9No/651-rewrite-help-for-list-clouds)

**Pull request merged**